hack/windows/Clean-AntreaNetwork.ps1

```diff
@@ -116,65 +78,32 @@ function CreateStartOVSvSwitchd() {
-    $Devs = $(Get-PnpDevice -Class net | ? Status -eq Unknown | Select InstanceId)
-    foreach ($Dev in $Devs) {
-        $RemoveKey = "HKLM:\SYSTEM\CurrentControlSet\Enum\$($Dev.InstanceId)"
-        Get-Item $RemoveKey | Select-Object -ExpandProperty Property | %{Remove-ItemProperty -Path $RemoveKey -Name $_ -Verbose }
+function clearOVSBridge() {
+    $ovsdbService =  Get-Service ovsdb-server -ErrorAction SilentlyContinue
+    if ( $ovsdbService -ne $null -and $ovsdbService.Status -eq "running") {
+        Write-Host "Delete OVS bridge from OVSDB: br-int"
+        ovs-vsctl.exe --no-wait --if-exists del-br $OVS_BR_ADAPTER
     }
 }
 
-function clearOVSBridge() {
-    $ovsStatus = $(Get-Service ovs-vswitchd).Status
-    if ("$ovsStatus" -EQ "running") {
-        Write-Host "Delete OVS bridge: br-int"
-        ovs-vsctl.exe --no-wait --if-exists del-br br-int
-        $MaxRetryCount = 10
-        $RetryCountRange = 1..$MaxRetryCount
-        $BrIntDeleted = $false
-        foreach ($RetryCount in $RetryCountRange) {
-            Write-Host "Waiting for OVS bridge deletion complete ($RetryCount/$MaxRetryCount)..."
-            $BrIntAdapter = $(Get-NetAdapter "$OVS_BR_ADAPTER" -ErrorAction SilentlyContinue)
-            if ($BrIntAdapter -eq $null) {
-                $BrIntDeleted = $true
-                break
-            }
-            if ($RetryCount -eq $MaxRetryCount) {
-                break
-            }
-            Start-Sleep -Seconds 5
-        }
-        if (!$BrIntDeleted) {
-            Write-Host "Failed to delete OVS Bridge, please retry the script or delete the bridge and HNS network manually."
-            return
+function ClearHnsNetwork() {
+    $vmSwitch = Get-VMSwitch -Name $AntreaHnsNetworkName -ErrorAction SilentlyContinue
+    if ($vmSwitch -ne $null) {
+        Write-Host "Remove vNICs"
+        Remove-VMNetworkAdapter -SwitchName $AntreaHnsNetworkName -ManagementOS -Confirm:$false -ErrorAction SilentlyContinue
+        $hnsNetwork = Get-HnsNetwork | ? Name -eq $AntreaHnsNetworkName
+        if ($hnsNetwork -ne $null) {
+            Write-Host "Remove HnsNetwork: $AntreaHnsNetworkName"
+            $uplink = $hnsNetwork.NetworkAdapterName
+            Get-HnsNetwork -Id $hnsNetwork.Id | Remove-HnsNetwork -ErrorAction Continue
+            Set-NetAdapterBinding -Name $uplink -ComponentID vms_pp -Enabled $false
         }
+        Remove-VMSwitch -Name $AntreaHnsNetworkName -Force -ErrorAction SilentlyContinue
     }
 }
 
-$BrIntDeleted = $(Get-NetAdapter "$OVS_BR_ADAPTER") -Eq $null
-if ($BrIntDeleted -eq $false) {
-    clearOVSBridge
-}
-$uplink = ""
-$AntreaHnsNetworkName = "antrea-hnsnetwork"
-$NetId = GetHnsnetworkId($AntreaHnsNetworkName)
-if ($NetId -ne $null) {
-    Write-Host "Remove HnsNetwork: $AntreaHnsNetworkName"
-    $uplink = $(Get-HnsNetwork -Id $NetId).NetworkAdapterName
-    Get-HnsNetwork -Id $NetId | Remove-HnsNetwork
-}
-
-RemoveNetworkAdapter $OVS_BR_ADAPTER
-RemoveNetworkAdapter "antrea-gw0"
```

Comment:
Could you add a commit message to clarify why vNic was removed from Windows VMSwitch and why netadapter such as antrea-gw0 doesn't need to be removed in script any more?

Reply:
added.